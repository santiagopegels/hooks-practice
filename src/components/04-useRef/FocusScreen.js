import React, { useRef } from 'react'

export const FocusScreen = () => {

    const inputRef = useRef()

    const handleClick = () => {
        inputRef.current.focus()
    }

    return (
        <div>
            <h1>Focus Screen</h1>
            <hr></hr>

            <input
                ref={inputRef}
                className="form-control"
                placeholder="Your Name"
            />

            <button
                className="btn btn-outline-primary mt-3"
                onClick={handleClick}
            >
                Focus
            </button>
        </div>
    )
}
