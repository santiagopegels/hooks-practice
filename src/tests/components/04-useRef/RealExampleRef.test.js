import React from 'react'
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef'
import { shallow } from 'enzyme'


describe('tests on <RealExampleRef />', () => {
    const wrapper = shallow(<RealExampleRef />)

    test('should show correctly', () => {

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false)
        
    })

    test('should show <MultipleCustomHooks />', () => {
        wrapper.find('button').simulate('click')
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true)
    })
})
