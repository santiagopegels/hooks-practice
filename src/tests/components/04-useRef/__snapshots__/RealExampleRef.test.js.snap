// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <RealExampleRef /> should show correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
